### PR TITLE
GEN-1322: API Entity - Remove Beta

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes:

Fixes GEN-1322

This PR includes changes for the following things
- Remove the beta tag from API services
- Support test connection for API Service Type REST
- Show Ingestion pipelines for API services
- Show Ingestion tab for API service details page

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/examples/workflows/rest.yaml [hunk 1]
```diff
@@ -3,9 +3,9 @@ source:
   serviceName: openapi_rest
   serviceConnection:
     config:
-      type: REST
+      type: Rest
```
Comment (harshach) [inline ingestion/src/metadata/examples/workflows/rest.yaml:6]: why are we changing type "Rest"?

Comment (Sachin-chaurasiya) [reply ingestion/src/metadata/examples/workflows/rest.yaml:6]: cc: @harshsoni2024

Comment (harshsoni2024) [reply ingestion/src/metadata/examples/workflows/rest.yaml:6]: Actually when we used service name in all caps, we get multiple errors in OM apis, airflow connections while doing ingestion from UI. for e.g. In the connection class [func](https://github.com/open-metadata/OpenMetadata/blob/f3c107025a722e09012e7912a264074f8a5acf4d/ingestion/src/metadata/ingestion/api/parser.py#L70), it parses service name in capitalized format. We had to apply patch on multiple parts for the rest service so we had followed same naming conventions as we did in all other dashboard, database connectors.

**Review by chirag-madlani — DISMISSED** [review]
Looks good UI side

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**